book-library/script.js

```diff
@@ -101,3 +101,109 @@ function render() {
     });
   }
 }
+*/
+
+
+let myLibrary = [];
+
+window.addEventListener("load", function (e) {
+  populateStorage();
+  render();
+});
+
+function populateStorage() {
+  if (myLibrary.length == 0) {
+    let book1 = new Book("Robison Crusoe", "Daniel Defoe", "252", true);
+    let book2 = new Book(
+      "The Old Man and the Sea",
+      "Ernest Hemingway",
+      "127",
+      true
+    );
+    myLibrary.push(book1);
+    myLibrary.push(book2);
+    render();
+  }
+}
+
+const title = document.getElementById("title");
+const author = document.getElementById("author");
+const pages = document.getElementById("pages");
+const check = document.getElementById("check");
+
+function submit() {
+  if (
+    title.value == null ||
+    title.value == "" ||
+    author.value == null ||
+    author.value == "" ||
+    pages.value == null ||
+    pages.value == ""
+  ) {
+    alert("Please fill all fields!");
+    return false;
+  } else {
+    let book = new Book(title.value, author.value, pages.value, check.checked);
+    myLibrary.push(book);
+    render();
+  }
+}
+
+function Book(title, author, pages, check) {
+  this.title = title;
+  this.author = author;
+  this.pages = pages;
+  this.check = check;
+}
+
+function render() {
+  let table = document.getElementById("display");
+  let rowsNumber = table.rows.length;
+  //delete old table
+  for (let n = rowsNumber - 1; n > 0; n--) {
+    table.deleteRow(n);
+  }
+  //insert updated row and cells
+  let length = myLibrary.length;
+  for (let i = 0; i < length; i++) {
+    let row = table.insertRow(1);
+    let cell1 = row.insertCell(0);
+    let cell2 = row.insertCell(1);
+    let cell3 = row.insertCell(2);
+    let cell4 = row.insertCell(3);
+    let cell5 = row.insertCell(4);
+    cell1.innerHTML = myLibrary[i].title;
+    cell2.innerHTML = myLibrary[i].author;
+    cell3.innerHTML = myLibrary[i].pages;
+
+    //add and wait for action for read/unread button
+    let changeBut = document.createElement("button");
+    changeBut.id = i;
+    changeBut.className = "btn btn-success";
+    cell4.appendChild(changeBut);
+    let readStatus = "";
+    if (myLibrary[i].check == false) {
+      readStatus = "No";
+    } else {
+      readStatus = "Yes";
+    }
+    changeBut.innerHTML = readStatus;
+
+    changeBut.addEventListener("click", function () {
+      myLibrary[i].check = !myLibrary[i].check;
+      render();
+    });
+
+    //add delete button to every row and render again
+    let delBut = document.createElement("button");
+    delBut.id = i + 5;
+    cell5.appendChild(delBut);
+    delBut.className = "btn btn-warning";
+    delBut.innerHTML = "Delete";
```

Comment:
Comment: Well done! 


Consider: Extra points. alternative to .innerHTML to set the button text is the textContent property. This is safer and more efficient because it doesn't parse HTML.